tests/ui/unnecessary_lazy_eval.rs

```diff
@@ -98,20 +96,9 @@ fn main() {
     let _: Result<usize, usize> = res.or_else(|err| Ok(err));
     let _: Result<usize, usize> = res.or_else(|err| Err(err));
 
-    // These are handled by bind_instead_of_map
+    // These are also handled by bind_instead_of_map
     let _: Result<usize, usize> = res.and_then(|_| Ok(2));
-    let _: Result<usize, usize> = res.and_then(|_| Ok(astronomers_pi));
-    let _: Result<usize, usize> = res.and_then(|_| Ok(ext_str.some_field));
-
     let _: Result<usize, usize> = res.and_then(|_| Err(2));
```

Comment:
Why are those test cases removed?

Also: linting the same code twice with different lints is problematic, since the suggested code can differ, which might be confusing.

Reply:
> Why are those test cases removed?

I dropped a couple of redundant test cases because of the test file line count limit imposed by the test suite. They all have counterparts above in the file. 

> Also: linting the same code twice with different lints is problematic, since the suggested code can differ, which might be confusing.

Should this lint be avoided in this multi-lint case? Is there maybe a priority system in case of overlapping lints?

Reply:
Hm, what is this line count limit?

In principal I personally would rather have too many testcases than too little since it can help spot issues in different lints as well.

Reply:
I've changed the `bind_instead_of_map` lint so it exposes its applicability to this lint. They should not overlap anymore, as implemented before these changes.